File: app/src/main/java/com/example/myfirstapp/utils/printer/SearchPrinter.java
package com.example.myfirstapp.utils.printer;
import android.content.Context;

import com.example.myfirstapp.utils.printer.wepoyprinter.PrinterWepoyManager;

public class SearchPrinter {
    public static PrinterToBoleta searchPrinter(Context ctx, PrinterToBoleta.IPrinterListener listener){
        return new PrinterWepoyManager(ctx, listener);
    }
}

File: app/src/main/java/com/example/myfirstapp/utils/printerK/SearchPrinter.kt
package com.example.myfirstapp.utils.printerK

import android.content.Context
import com.example.myfirstapp.utils.printerK.wepoyprinter.PrinterRipleyManager

object SearchPrinter {
    fun searchPrinter(ctx: Context, listener: PrinterToTicket.IPrinterListener): PrinterToTicket {
        return PrinterRipleyManager(ctx, listener)
    }
}